pkg/operator/statefulset.go

```diff
@@ -24,74 +24,154 @@ import (
 
 const configReloaderPort = 8080
 
+type ConfigReloader struct {
+	name           string
+	config         ReloaderConfig
+	reloadURL      url.URL
+	listenLocal    bool
+	localHost      string
+	logFormat      string
+	logLevel       string
+	additionalArgs []string
+	volumeMounts   []v1.VolumeMount
+	shard          int32
+}
+
+type option = func(*ConfigReloader)
+
+func ReloaderRunOnce() option {
+	return func(c *ConfigReloader) {
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--watch-interval=%d", 0))
+	}
+}
+
+func ListenAddress() option {
+	return func(c *ConfigReloader) {
+		var listenAddress string
+		if c.listenLocal {
+			listenAddress = c.localHost
+		}
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--listen-address=%s:%d", listenAddress, configReloaderPort))
+	}
+}
+
+func ReloadURL() option {
```

Comment:
I would remove this function and move the logic to `NewConfigReloader()` which could add the `--reload-url` argument if c.reloadURL.String() isn't empty.